test/src/unittests/rhythm/test_loopbpmconfidence.py

```diff
@@ -0,0 +1,157 @@
+         then this will indicate that accuracy of algorithm has changed.
+         The 3rd param in assertAlmostEqual() has been optimised, lowest possible value chosen.
+        """
+
+        # Randomly chosen points, Section 1
+        samplePoint1 = 1000
+        samplePoint2 = 6000
+
+        bpmEstimate = 125
+        expectedConfidence = 0.53
+        confidence = LoopBpmConfidence()(audio[samplePoint1:samplePoint2], bpmEstimate)                
+        self.assertAlmostEqual(expectedConfidence, confidence, 0.02) 
+
+        bpmEstimate = 120
+        expectedConfidence = 0.55
+        confidence = LoopBpmConfidence()(audio[samplePoint1:samplePoint2], bpmEstimate)                
+        self.assertAlmostEqual(expectedConfidence, confidence, 0.02) 
+        
+        bpmEstimate = 70
+        expectedConfidence = 0.74
+        confidence = LoopBpmConfidence()(audio[samplePoint1:samplePoint2], bpmEstimate)            
+        self.assertAlmostEqual(expectedConfidence, confidence, 0.02) 
+
+        # Randomly chosen points, Section 2
+        samplePoint1 = 22000
+        samplePoint2 = 38000
+        expectedEstimate = 100
+        
+        bpmEstimate = 125
+        expectedConfidence = 0.51
+        confidence = LoopBpmConfidence()(audio[samplePoint1:samplePoint2], bpmEstimate)                
+        self.assertAlmostEqual(expectedConfidence, confidence, 0.02) 
+
+        bpmEstimate = 120
+        expectedConfidence = 0.45
+        confidence = LoopBpmConfidence()(audio[samplePoint1:samplePoint2], bpmEstimate)                
+        self.assertAlmostEqual(expectedConfidence, confidence, 0.02) 
+        
+        bpmEstimate = 70
+        expectedConfidence = 0.15
+        confidence = LoopBpmConfidence()(audio[samplePoint1:samplePoint2], bpmEstimate)            
+        self.assertAlmostEqual(expectedConfidence, confidence, 0.05) 
+        
+    def testEmpty(self):
+        # Zero estimate check results in zero confidence
+        emptyAudio = []
+        bpmEstimate = 0
+        confidence = LoopBpmConfidence()(emptyAudio, bpmEstimate)                  
+        self.assertEquals(0, confidence)
+
+        # Non-zero estimate check results in zero confidence
+        emptyAudio = []
+        bpmEstimate = 125
+        confidence = LoopBpmConfidence()(emptyAudio, bpmEstimate)                  
+        self.assertEquals(0, confidence)
+
+    def testZero(self):
+        zeroAudio = zeros(1000)
+        bpmEstimate = 0
+        confidence = LoopBpmConfidence()(zeroAudio, bpmEstimate)
+        self.assertEquals(0, confidence)
+
+        # Non-zero estimate check results in non-zero confidence
+        # FIX-ME: Maybe given contant zero input, we should expect zero confidence
+        bpmEstimate = 125
+        confidence = LoopBpmConfidence()(zeroAudio, bpmEstimate)
+        self.assertNotEquals(0, confidence)
+         
+    def testConstantInput(self):
+        onesAudio = ones(100)
```

Comment:
Let's use the audio length consistent with testZero. Also, in both tests, 1000 samples is a too small value for BPM estimation. Let's set it to something meaningful like 2 seconds (which is a full bar in 120 BPM)